test/unit/eigensolver/test_reduction_to_band.cpp

```diff
@@ -396,7 +396,6 @@ TYPED_TEST(ReductionToBandTestMC, CorrectnessDistributed) {
       copy(reference, matrix_a);
 
       auto local_taus = eigensolver::reductionToBand<Backend::MC>(comm_grid, matrix_a);
-      pika::threads::get_thread_manager().wait();
 
       checkUpperPartUnchanged(reference, matrix_a);
 
```

Comment:
To do: `allGatherTaus` below does blocking communication as well. Is it making this hang?

Reply:
Part of the puzzle, but not the full answer. The blocking communication has been shipped off to a separate thread now.